### PR TITLE
First parts of UI for the federation project on FE2

### PR DESCRIPTION
**Pull request by quadcopterman**
Added the first step of the federation project with some menus and other UI elements.

Here are some screenshots of the new UI elements :
image 
image
image
image

### CODE REVIEW

**Review by matteosz — COMMENTED** [review]
Very good job so far, clean code and well documented! Just a question about the gradle change. 

Also when you're done with the PR don't forget to add to the PR the screenshots with the new UIs 😄

**Comment by matteosz** [comment]
Quick reminder - As I commented in this old [PR](https://github.com/dedis/popstellar/pull/1758#:~:text=Just%20a%20reminder,careful%20about%20it.), try to keep number of commits low and try to avoid pushing useless commits such as "ktfmt", "formatting" (squash them before pushing)

File: fe2-android/build.gradle [hunk 1]
```diff
@@ -21,6 +21,7 @@ buildscript {
 
 plugins {
     id 'com.android.application' version '7.4.2' apply false
+    id 'org.jetbrains.kotlin.android' version '1.8.22' apply false
```
[the remark this answers is not in the document]
Comment (quadcopterman) [reply fe2-android/build.gradle:24]: Actually I don't know. It was automatically added by Android Studio, I'll try to remove it and see if it breaks something :)

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/federation/LinkedOrganizationsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  private var invitationPage =
+      View.OnClickListener {
+        LaoActivity.setCurrentFragment(
+            parentFragmentManager, R.id.fragment_linked_organizations_invite) {
+              LinkedOrganizationsInviteFragment.newInstance(true)
+            }
+      }
+
+  private fun handleBackNav() {
+    LaoActivity.addBackNavigationCallbackToEvents(requireActivity(), viewLifecycleOwner, TAG)
```
[the remark this answers is not in the document]
Comment (quadcopterman) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/federation/LinkedOrganizationsFragment.kt:89]: This is actually normal, we are in the Linked Organizations section and we hit the back button we go in Events (it's the case for all the other sections as well, we always arrive in Event when we go back)

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/federation/LinkedOrganizationsInviteFragment.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  override fun onResume() {
+    super.onResume()
+    laoViewModel.setPageTitle(R.string.invite_other_organization)
+    laoViewModel.setIsTab(false)
+  }
+
+  private fun handleBackNav() {
+    LaoActivity.addBackNavigationCallbackToEvents(requireActivity(), viewLifecycleOwner, TAG)
```
[the remark this answers is not in the document]
Comment (quadcopterman) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/federation/LinkedOrganizationsInviteFragment.kt:74]: Yes this is a problem as we are in invite and it should go back to Linked Organizations. I'll fix it

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/federation/LinkedOrganizationsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        binding.noOrganizationsText.setText(R.string.no_organizations_organizer_text)
+        binding.addLinkedOrganization.visibility = View.VISIBLE
+      } else {
+        binding.noOrganizationsText.setText(R.string.no_organizations_non_organizer_text)
+        binding.addLinkedOrganization.visibility = View.GONE
+      }
+    }
+
+    binding.addLinkedOrganization.setOnClickListener(observeButton)
+    binding.inviteOtherOrganization.setOnClickListener(invitationPage)
```
[the remark this answers is not in the document]
Comment (quadcopterman) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/federation/LinkedOrganizationsFragment.kt:50]: Yes, parts of the UI are not implemented yet

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/MainMenuTab.kt [hunk 5]
```diff
@@ -11,6 +11,7 @@ enum class MainMenuTab(@get:IdRes val menuId: Int) {
   POPCHA(R.id.main_menu_popcha),
   WITNESSING(R.id.main_menu_witnessing),
   TOKENS(R.id.main_menu_tokens),
+  LINKED_ORGANIZATIONS(R.id.main_menu_linked_organisations),
```
[the remark this answers is not in the document]
Comment (quadcopterman) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/MainMenuTab.kt:14]: Thanks, I'll correct  that

**Comment by quadcopterman** [comment]
I corrected the typos and the errors. For the tests, I agree that 65% coverage is not that high but I already wrote tests for all of the UI components, so I don't really know what to add (and UI tests are hell to setup, as you mentioned). Also most of the uncovered lines are "Partially covered lines".

**Review by Kaz-ookid — APPROVED** [review]
LGTM after the changes, some other suggestions are still up but they might as well be done some time later in another PR :) !

**Review by matteosz — APPROVED** [review]
Impressive work!

Agree with some of the comments made by @Kaz-ookid (increase further test coverage and improve abstraction of some animation/ui functions putting them into a util file). I've reviewed most of the PR when it was a draft, but now it's much cleaner and better tested, so I'm happy to approve it as is and then you can address those comments in a next PR, since this one is just the foundation of the federation work :)

**Comment by quadcopterman** [comment]
Thanks :)

**Pull request merged**